src/main/java/org/kohsuke/github/GHCheckRunBuilder.java

```diff
@@ -0,0 +1,370 @@
+     * @throws IOException
+     *             for the usual reasons
+     */
+    public @NonNull GHCheckRun create() throws IOException {
+        List<Annotation> extraAnnotations;
+        if (output != null && output.annotations != null && output.annotations.size() > MAX_ANNOTATIONS) {
+            extraAnnotations = output.annotations.subList(MAX_ANNOTATIONS, output.annotations.size());
+            output.annotations = output.annotations.subList(0, MAX_ANNOTATIONS);
+        } else {
+            extraAnnotations = Collections.emptyList();
+        }
+        GHCheckRun run = requester.with("output", output).with("actions", actions).fetch(GHCheckRun.class).wrap(repo);
+        while (!extraAnnotations.isEmpty()) {
+            Output output2 = new Output(null, output.title, output.summary);
+            int i = Math.min(extraAnnotations.size(), MAX_ANNOTATIONS);
+            output2.annotations = extraAnnotations.subList(0, i);
+            extraAnnotations = extraAnnotations.subList(i, extraAnnotations.size());
+            run = repo.root.createRequest()
+                    .withPreview(Previews.ANTIOPE)
+                    .method("PATCH")
+                    .with("output", output2)
+                    .withUrlPath(repo.getApiTailUrl("check-runs/" + run.id))
+                    .fetch(GHCheckRun.class)
+                    .wrap(repo);
+        }
+        return run;
+    }
+
+    /**
+     * @see <a href="https://developer.github.com/v3/checks/runs/#output-object">documentation</a>
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static final class Output {
+
+        private final transient GHCheckRunBuilder builder;
+        private final String title;
+        private final String summary;
+        private String text;
+        List<Annotation> annotations;
+        List<Image> images;
+
+        Output(GHCheckRunBuilder builder, String title, String summary) {
+            this.builder = builder;
+            this.title = title;
+            this.summary = summary;
+        }
+
+        public @NonNull Output withText(@CheckForNull String text) {
+            this.text = text;
+            return this;
+        }
+
+        /**
+         * Drafts a single-line annotation section.
+         *
+         * @param path
+         *            as in GitHub documentation
+         * @param line
+         *            {@code startLine} and {@code endLine} together
+         * @param annotationLevel
+         *            as in GitHub documentation
+         * @param message
+         *            as in GitHub documentation
+         * @return use {@link Annotation#done} to continue
+         */
+        public @NonNull Annotation withAnnotation(@NonNull String path,
+                int line,
+                @NonNull GHCheckRun.AnnotationLevel annotationLevel,
+                @NonNull String message) {
+            return withAnnotation(path, line, line, annotationLevel, message);
```

Comment:
Max limit for annotations ?

Reply:
Nope, >50 is automatically handled, see above.